src/tiled/createobjecttool.h

```diff
@@ -66,6 +69,8 @@ class CreateObjectTool : public AbstractObjectTool
     ObjectGroupItem *mObjectGroupItem;
     MapObjectItem *mNewMapObjectItem;
     MapObjectItem *mOverlayPolygonItem;
+    bool mExtending;
+    bool mExtendingFirst;
```

Comment:
Well, I missed it too, but this is obviously because we're in `CreatePolylineObjectTool::finishNewMapObject` there, so we should at least avoid calling that exact function recursively.